Backend/src/main/java/ca/mcgill/cooperator/service/ReportSectionConfigService.java

```diff
@@ -146,11 +178,21 @@ public ReportSectionConfig deleteReportSectionConfig(ReportSectionConfig rsConfi
 
         // first delete from parent ReportConfig
         ReportConfig reportConfig = rsConfig.getReportConfig();
-        Set<ReportSectionConfig> rsConfigs = reportConfig.getReportSectionConfigs();
+        Set<ReportSectionConfig> rsConfigs = new HashSet<>(reportConfig.getReportSectionConfigs());
```

Comment:
should we be doing all list/set instantiation like this? whats the benefit of adding the hashset part vs just doing `reportConfig.getReportSectionConfigs()`

Reply:
v important because of pointer issue

Reply:
otherwise its gonna clear it when you do the set in the next line since they point to each other

Reply:
Yup exactly that, it's a sneaky bug